### PR TITLE
fixing mobile password validation for servers with draconian policy

### PR DESCRIPTION
**Pull request by calellowitz**
http://manage.dimagi.com/default.asp?237866#1222615
@esoergel @sravfeyn cc: @gcapalbo 
This pr addresses a series of related errors with setting mobile worker passwords on the india server which uses a special password strength validator. Most of the issues were caused by the js incorrectly marking weak passwords as strong based on differences in the engine it was using. The changes do not attempt to validate passwords but make the guidelines clearer when those settings are on.

### CODE REVIEW

File: corehq/apps/users/static/users/js/mobile_workers.ng.js [hunk 1]
```diff
@@ -315,7 +322,7 @@
         };
     };
 
-    mobileWorkerDirectives.validatePassword = function ($http, $q, djangoRMI) {
+    mobileWorkerDirectives.validatePasswordStandard = function ($http, $q, djangoRMI) {
```
Comment (dimagimon) [inline corehq/apps/users/static/users/js/mobile_workers.ng.js]: 'djangoRMI' is defined but never used (no-unused-vars)

File: corehq/apps/users/static/users/js/mobile_workers.ng.js [hunk 2]
```diff
@@ -341,6 +348,30 @@
         };
     };
 
+    mobileWorkerDirectives.validatePasswordDraconian = function ($http, $q, djangoRMI) {
```
Comment (dimagimon) [inline corehq/apps/users/static/users/js/mobile_workers.ng.js]: 'djangoRMI' is defined but never used (no-unused-vars)

File: corehq/apps/users/static/users/js/mobile_workers.ng.js [hunk 3]
```diff
@@ -341,6 +348,30 @@
...
+                        .addClass('has-pending');
+                    if ($formElements.password().hasClass('non-default')) {
+                        $formElements.passwordHint()
+                            .text(gettext("Password Requirements: 1 special character, " +
+                                          "1 number, 1 capital letter, minimum length of 8 characters."));
+                    }
+
+                    return true;
+                };
+            }
```
Comment (dimagimon) [inline corehq/apps/users/static/users/js/mobile_workers.ng.js]: Missing trailing comma. (comma-dangle)

**Comment by gcapalbo** [comment]
:+1: on :white_check_mark:

**Pull request merged**